### ISSUE TITLE
[New build - DEFAULT] 20w230a: Maglock system.

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#2687]
The maglock system block is added, you can use it to carry other entity along the main one.
Active the block to trigger the lock, deactivate it to release it.

New Features:
- Maglock and Maglock, small blocks added.

Bugfixes:
#2684 Massive FPS drop when tile has a high number of decals.
#2686 Wrong Heat surface computation.

**Comment by BrickMacklin** [comment]
As soon as I start placing blocks on a larger build the FPS drops to zero for several seconds

**Comment by tsunamayo** [comment]
@BrickMacklin it was fine on 20w29d? Thanks

**Comment by BrickMacklin** [comment]
@tsunamayo Yes. There were zero issues on there. Several others on the Discord are reporting the same thing with larger builds.

**Comment by alxnns1** [comment]
I am also experiencing this lag when placing bricks on large builds

**Comment by tsunamayo** [comment]
ah okay I see why, I will repush a build asap. Thanks

**Comment by tsunamayo** [comment]
new build is live, sorry for the miss guys!

**Comment by Kaiser-Indrasil** [comment]
Now we only need persistent mag-locks, so we could spawn ships with swappable turrets and what-not already attached. <3

**Comment by tsunamayo** [comment]
you mean an export blueprint with entity attached? Right now maglock will be persistent on save reload.

**Comment by Kaiser-Indrasil** [comment]
Yes, that is exactly what I meant. However, now we will be able to build freighters with detachable cargo, for example, and I love it!

**Comment by ProPeach** [comment]
Is the ability to unlock a maglock block planned? Right now they're permanent locks, I'm not sure if that's a bug of just a planned future feature
